device/core/devdoc/device_client_requirements.md

```diff
@@ -272,3 +272,15 @@ interface DeviceMethodEventHandler {
 #### disconnect
 
 **SRS_NODE_DEVICE_CLIENT_16_019: [** The `disconnect` event shall be emitted when the client is disconnected from the server. **]**
+
+**SRS_NODE_DEVICE_CLIENT_16_097: [** If the transport emits a `disconnect` event the client while subscribed to C2D messages the retry policy shall try to re-enable the C2D functionality using the transport `enableC2D` method. **]**
```

Comment:
Nice!  These requirements will also need to apply to inputMessage events.  

side note: I wonder if there's a way to re-write these requirements so the text is only written once but it applies to all 3 (soon 4) features.  I'm sure the test will use a forEach to test these requirements.  I wonder if the requirements could be written in the same way.  Maybe instead of grouping by method, we should group by sub-feature, and have functionality that applies to sub-features (if that makes sense).  Just thinking it might make for better req docs with less copy/paste (says the guy who just did a bunch of copy/paste in this doc).

Reply:
well it also depends if this group of methods would always evolve together. i'd be in favor of factoring in requirements though.

---
In reply to: [171948837](https://github.com/pierreca/azure-iot-sdk-node/pull/2#discussion_r171948837) [](ancestors = 171948837)